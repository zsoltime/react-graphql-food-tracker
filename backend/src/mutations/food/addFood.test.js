'use strict';

const addFood = require('./addFood');

describe('addFood mutation', () => {
  it('inserts food to the database and returns it', async () => {
    const args = {
      brand: 'Meridian',
      calories: 596,
      carb: 11.6,
      ean: '5060132283424',
      fat: 46,
      fatSaturated: 8.2,
      fibre: 8.5,
      name: 'Crunchy peanut butter',
      protein: 29.6,
      servingSize: 100,
      servingUnit: 'G',
      sugar: 5.9,
    };
    const mockedContext = {
      db: {
        mutation: {
          createFood: jest
            .fn()
            .mockImplementation(() => Promise.resolve({ ...args, user: 'validUserId' })),
        },
      },
      request: { userId: 'validUserId' },
    };

    const food = await addFood(
      null,
      args,
      mockedContext,
      'mockedInfoObject'
    );

    expect(mockedContext.db.mutation.createFood).toBeCalled();
    expect(
      mockedContext.db.mutation.createFood.mock.calls[0]
    ).toMatchSnapshot();
    expect(food).toMatchSnapshot();
  });

  it('throws error if user not signed in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      addFood(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
