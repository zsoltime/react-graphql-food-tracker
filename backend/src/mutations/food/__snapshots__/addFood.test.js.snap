// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addFood mutation inserts food to the database and returns it 1`] = `
Array [
  Object {
    "data": Object {
      "brand": "Meridian",
      "calories": 596,
      "carb": 11.6,
      "ean": "5060132283424",
      "fat": 46,
      "fatSaturated": 8.2,
      "fibre": 8.5,
      "name": "Crunchy peanut butter",
      "protein": 29.6,
      "servingSize": 100,
      "servingUnit": "G",
      "sugar": 5.9,
      "user": Object {
        "connect": Object {
          "id": "validUserId",
        },
      },
    },
  },
  "mockedInfoObject",
]
`;

exports[`addFood mutation inserts food to the database and returns it 2`] = `
Object {
  "brand": "Meridian",
  "calories": 596,
  "carb": 11.6,
  "ean": "5060132283424",
  "fat": 46,
  "fatSaturated": 8.2,
  "fibre": 8.5,
  "name": "Crunchy peanut butter",
  "protein": 29.6,
  "servingSize": 100,
  "servingUnit": "G",
  "sugar": 5.9,
  "user": "validUserId",
}
`;

exports[`addFood mutation throws error if user not signed in 1`] = `"You must be signed in to add foods"`;
